functions/src/SQR.functions.js

```diff
@@ -20,33 +18,64 @@ const SQR_Helper = require('./SQR_Helper');
 
 exports.updateFilesOnNewAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
     .onCreate((snapshot, context) => {
-        return SQR_Helper.UpdateFilesOnNewAllotment(
-                snapshot, 
-                db,
-                helpers.updateFile
-            );
+        const original = snapshot.val();
+        let newDocKey = snapshot.key;
+        original.files.forEach(file => {
+            let file_ref = db.ref(`/sqr/files/${original.list}/${file}`);
+            file_ref.child("status").once('value')
+            .then(snapshot => {
+                if (snapshot.exists())
+                    file_ref.update(
+                        {
+                            status: 'Given',
+                            allotment: {
+                                timestampGiven: new Date().getTime(),
+                                timestampDone: null,
+                                devotee: original.devotee
+                            }
+                        }, err => {
+                            if (!err)
+                                db.ref(`/sqr/allotments/${newDocKey}`).update({ filesAlloted: true });
+                        });
+                return 1;
+            }).catch(err => console.log(err));
+        });
+
+        return 1;
 });
 
 exports.sendEmailOnFileAllotment = functions.database.ref('/sqr/allotments/{allotment_id}')
     .onUpdate((change, context) => {
         const old = change.before.val();
         const _new = change.after.val();        
```

Comment:
I don't like the underscore here. And something more obvious would be better, like `newAllotment` or at least `newData`.

Reply:
ok